[Issue]: Building TheRock with `-DTHEROCK_BUNDLE_SYSDEPS=OFF` fails

With #195 landed, which re-introduced `lib/rocm_sysdeps/` to the path in
https://github.com/ROCm/TheRock/blob/939ab298a2ea2cd77fbe7a204814876c963c81c3/base/post_hook_rocm_smi_lib.cmake#L1-L2
building TheRock fails when `THEROCK_BUNDLE_SYSDEPS` set to `OFF` fails with:
```
[hipBLASLt configure] CMake Error in clients/gtest/CMakeLists.txt:
[hipBLASLt configure]   Imported target "rocm_smi64" includes non-existent path
[hipBLASLt configure]
[hipBLASLt configure]     "TheRock/build/base/rocm_smi_lib/dist/lib/rocm_sysdeps/include"
[hipBLASLt configure]
[hipBLASLt configure]   in its INTERFACE_INCLUDE_DIRECTORIES.  Possible reasons include:
[hipBLASLt configure]
[hipBLASLt configure]   * The path was deleted, renamed, or moved to another location.
[hipBLASLt configure]
[hipBLASLt configure]   * An install or uninstall procedure did not complete successfully.
[hipBLASLt configure]
[hipBLASLt configure]   * The installation package was faulty and references files it does not
[hipBLASLt configure]   provide.
[hipBLASLt configure]
```

--- a/README.md
+++ b/README.md
@@ -26,7 +26,7 @@ pip install -r requirements.txt
 Dev tools:
 
 ```
-sudo apt install gfortran git-lfs ninja-build cmake g++ pkg-config xxd libgtest-dev patchelf automake
+sudo apt install gfortran git-lfs ninja-build cmake g++ pkg-config xxd libgmock-dev libgtest-dev patchelf automake
 ```
 
 ## On Windows

--- a/base/CMakeLists.txt
+++ b/base/CMakeLists.txt
@@ -52,6 +52,10 @@ if(NOT WIN32)  # TODO(#36): Enable on Windows and/or make subproject inclusion g
 
 therock_cmake_subproject_declare(rocm_smi_lib
   EXTERNAL_SOURCE_DIR "rocm_smi_lib"
+  CMAKE_ARGS
+    # See the post_hook, which needs to advertise install interface directories
+    # in this case.
+    "-DTHEROCK_HAS_BUNDLED_LIBDRM=${THEROCK_BUNDLED_LIBDRM}"
   INTERFACE_LINK_DIRS
     "lib"
   RUNTIME_DEPS

--- a/base/post_hook_rocm_smi_lib.cmake
+++ b/base/post_hook_rocm_smi_lib.cmake
@@ -1,6 +1,15 @@
-target_include_directories(rocm_smi64 PUBLIC
-  "$<INSTALL_INTERFACE:lib/rocm_sysdeps/include>"
-)
-target_include_directories(oam PUBLIC
-  "$<INSTALL_INTERFACE:lib/rocm_sysdeps/include>"
-)
+# smi_lib public headers include DRM, so in the bundled case, we must ensure
+# that the installed libraries add the include directory. This is because
+# libdrm does not cross the install boundary as a dep (since it is pkgconfig
+# and was not set up that way).
+# Conversely, if we include directories that do not exist, CMake consumers
+# will error out due to the missing directories.
+# See the project declaration for where this flag is defined.
+if(THEROCK_HAS_BUNDLED_LIBDRM)
+  target_include_directories(rocm_smi64 PUBLIC
+    "$<INSTALL_INTERFACE:lib/rocm_sysdeps/include>"
+  )
+  target_include_directories(oam PUBLIC
+    "$<INSTALL_INTERFACE:lib/rocm_sysdeps/include>"
+  )
+endif()

--- a/base/post_hook_rocm_smi_lib.cmake
+++ b/base/post_hook_rocm_smi_lib.cmake
@@ -1,4 +1,4 @@
-# smi_lib public headers include DRM, so in the bundled case, we must ensure
+# smi_lib public headers include libdrm, so in the bundled case, we must ensure
 # that the installed libraries add the include directory. This is because
 # libdrm does not cross the install boundary as a dep (since it is pkgconfig
 # and was not set up that way).

